core/src/main/java/org/apache/iceberg/rest/requests/UpdateRequirementParser.java

```diff
@@ -0,0 +1,264 @@
+      case ASSERT_REF_SNAPSHOT_ID:
+        return readAssertRefSnapshotId(jsonNode);
+      case ASSERT_LAST_ASSIGNED_FIELD_ID:
+        return readAssertLastAssignedFieldId(jsonNode);
+      case ASSERT_LAST_ASSIGNED_PARTITION_ID:
+        return readAssertLastAssignedPartitionId(jsonNode);
+      case ASSERT_CURRENT_SCHEMA_ID:
+        return readAssertCurrentSchemaId(jsonNode);
+      case ASSERT_DEFAULT_SPEC_ID:
+        return readAssertDefaultSpecId(jsonNode);
+      case ASSERT_DEFAULT_SORT_ORDER_ID:
+        return readAssertDefaultSortOrderId(jsonNode);
+      default:
+        throw new UnsupportedOperationException(
+            String.format("Unrecognized update requirement. Cannot convert to json: %s", type));
+    }
+  }
+
+  private static void writeAssertTableUUID(UpdateRequirement.AssertTableUUID requirement, JsonGenerator gen)
+      throws IOException {
+    gen.writeStringField(UUID, requirement.uuid());
+  }
+
+  private static void writeAssertRefSnapshotId(UpdateRequirement.AssertRefSnapshotID requirement, JsonGenerator gen)
+      throws IOException {
+    gen.writeStringField(NAME, requirement.refName());
+    gen.writeNumberField(SNAPSHOT_ID, requirement.snapshotId());
+  }
+
+  private static void writeAssertLastAssignedFieldId(UpdateRequirement.AssertLastAssignedFieldId requirement,
+      JsonGenerator gen) throws IOException {
+    gen.writeNumberField(LAST_ASSIGNED_FIELD_ID, requirement.lastAssignedFieldId());
+  }
+
+  private static void writeAssertLastAssignedPartitionId(UpdateRequirement.AssertLastAssignedPartitionId requirement,
+      JsonGenerator gen) throws IOException {
+    gen.writeNumberField(LAST_ASSIGNED_PARTITION_ID, requirement.lastAssignedPartitionId());
+  }
+
+  private static void writeAssertCurrentSchemaId(UpdateRequirement.AssertCurrentSchemaID requirement,
+      JsonGenerator gen) throws IOException {
+    gen.writeNumberField(SCHEMA_ID, requirement.schemaId());
+  }
+
+  private static void writeAssertDefaultSpecId(UpdateRequirement.AssertDefaultSpecID requirement, JsonGenerator gen)
+      throws IOException {
+    gen.writeNumberField(SPEC_ID, requirement.specId());
+  }
+
+  private static void writeAssertDefaultSortOrderId(UpdateRequirement.AssertDefaultSortOrderID requirement,
+      JsonGenerator gen) throws IOException {
+    gen.writeNumberField(SORT_ORDER_ID, requirement.sortOrderId());
+  }
+
+  @SuppressWarnings("unused")  // Keep same signature in case this requirement class evolves and gets fields
+  private static UpdateRequirement readAssertTableDoesNotExist(JsonNode node) {
+    return new UpdateRequirement.AssertTableDoesNotExist();
+  }
+
+  private static UpdateRequirement readAssertTableUUID(JsonNode node) {
+    String uuid = JsonUtil.getString(UUID, node);
+    return new UpdateRequirement.AssertTableUUID(uuid);
+  }
+
+  private static UpdateRequirement readAssertRefSnapshotId(JsonNode node) {
+    String name = JsonUtil.getStringOrNull(NAME, node);
```

Comment:
Isn't `ref` (or `name`) required? I would expect this to not allow `null`.

Reply:
Looking at the class definition, there are `null` checks with an else statement. That's what I based it on:

https://github.com/apache/iceberg/blob/a19ac7835c8e251fd6a042e056f29ee63262adec/core/src/main/java/org/apache/iceberg/rest/requests/UpdateTableRequest.java#L279-L297

Reply:
Oh wait I read that wrong from when I renamed it to `ref` temporarily. You're right.